Homeworks/Homework W8_ClassWork_HW/HomeworkClassroom.sql

```diff
@@ -0,0 +1,258 @@
+VALUES('2022-01-01', '2025-01-01', 2);
+INSERT INTO Contracts
+VALUES('2012-01-01', '2031-01-01', 1);
+INSERT INTO Contracts
+VALUES('2019-01-01', '2017-01-01', 2);
+
+SELECT *
+FROM Contracts;
+
+SELECT *
+FROM Customers;
+SELECT *
+FROM Contracts;
+
+INSERT INTO Customers_Contracts
+VALUES(1, 1);
+INSERT INTO Customers_Contracts
+VALUES(1, 2);
+INSERT INTO Customers_Contracts
+VALUES(2, 3);
+INSERT INTO Customers_Contracts
+VALUES(3, 4);
+INSERT INTO Customers_Contracts
+VALUES(3, 5);
+INSERT INTO Customers_Contracts
+VALUES(4, 6);
+INSERT INTO Customers_Contracts
+VALUES(4, 7);
+INSERT INTO Customers_Contracts
+VALUES(4, 8);
+
+SELECT *
+FROM Customers_Contracts;
+
+INSERT INTO Bills
+VALUES(234,'2022-12-25',  1, 1);
+INSERT INTO Bills
+VALUES(567,'2021-12-25',  1, 2);
+INSERT INTO Bills
+VALUES(12,'2020-12-25',  0, 2);
+INSERT INTO Bills
+VALUES(654,'2023-12-25',  0, 3);
+INSERT INTO Bills
+VALUES(98,'2019-11-25',  0, 4);
+INSERT INTO Bills
+VALUES(23,'2024-10-25',  0, 5);
+INSERT INTO Bills
+VALUES(243,'2023-10-25',  0, 6);
+INSERT INTO Bills
+VALUES(223,'2022-09-25',  1, 7);
+INSERT INTO Bills
+VALUES(2356,'2023-02-25',  1, 8);
+INSERT INTO Bills
+VALUES(2783,'2028-01-25',  0, 8);
+
+SELECT *
+FROM Bills;
+
+-- 3. Display all the customers' information ordered by the last name descending.
+SELECT *
+FROM Customers
+ORDER BY LastName DESC;
+
+-- 4. Display information about the bill with the highest amount.
+SELECT * 
+FROM Bills
+WHERE Amount = (SELECT MAX(Amount) FROM Bills);
+
+-- 5. Display the customers' information (just customer id, email address, street name, street number and city) togheter with their corresponding addresses.
+SELECT c.Id, c.Email, a.StreetName, a.StreetNumber, a.City
+FROM Customers c
+INNER JOIN Addresses a ON c.Id = a.CustomerId;
+
+-- 6. Pretend that one bill was paid today. Update it's corresponding information to reflect that.
+UPDATE Bills 
+SET IsPaid = 1
+WHERE Id = 9;
+
+SELECT *
+FROM Bills;
+
+-- 7. Display the contract's information (contract id, start date, end date, bill id, amount, deadline, isPaid) togheter with their corresponding bills.
+SELECT c.Id, c.StartDate, c.EndDate, b.Id, b.Amount, b.Deadline, b.IsPaid
+FROM Contracts c
+INNER JOIN Bills b ON c.Id = b.ContractId; 
+
+-- 8. Display the contract's information (contract id, start date, end date, contract type, bill id, amount, deadline, isPaid) togheter with their corresponding bills.
+SELECT c.Id, c.StartDate, c.EndDate, ct.Name, b.Id, b.Amount, b.Deadline, b.IsPaid
+FROM Contracts c
+INNER JOIN Bills b ON c.Id = b.ContractId
+JOIN ContractTypes ct on c.ContractType = ct.Id;
+
+-- 9. Display the customers' information (just customer id, email address, contract end date & contract type) togheter with their corresponding contract type(s).
+SELECT c.Id AS CustomerId, c.Email,t.EndDate AS ContractEndDate, t.ContractType
+FROM Customers c
+INNER JOIN Customers_Contracts o ON c.Id=o.CustomerId
+JOIN Contracts t  ON o.ContractId= t.Id;
```

Comment:
Another join with ContractTypes table would have produces more user friendly results so that instead of seeing contract type ids you would have seen the actual values ('gas', 'electricity'). From this (your query)
![image](https://user-images.githubusercontent.com/117767491/215324694-754aed75-e5a4-4bc3-9bf0-9b985c77c6db.png)
to this (query in the following comment):
![image](https://user-images.githubusercontent.com/117767491/215324721-b3f5dae3-69b7-43eb-b10a-94996637e32b.png)

Reply:
```

SELECT cu.Id, cu.Email, c.EndDate, ct.Name
FROM Customers cu 
JOIN Customers_Contracts cc ON cu.ID = cc.CustomerId
JOIN Contracts c ON cc.ContractId = c.Id 
JOIN ContractTypes ct ON c.ContractType = ct.Id;
```